CONTRIBUTING.txt

```diff
@@ -0,0 +1,98 @@
+    8. A PR MUST compile cleanly and pass project self-tests on all target platforms.
+    9. Each commit message in a PR MUST succinctly explain what the commit achieves.
+    10. The PR Description MUST consist of a single short (less than 50 characters) line stating the problem (“Problem: …") being solved, followed by a blank line and then the proposed solution (“Solution: …").
+    11. A “Valid PR” is one which satisfies the above requirements.
+
+6. Process
+
+    1. Change on the project follows the pattern of accurately identifying problems and applying minimal, accurate solutions to these problems.
+    2. To request changes, a user (end user, contributor, maintainer, or administrator) logs an issue on the project Platform issue tracker.
+    3. The user or contributor should write the issue by describing the problem they face or observe. Links to the forum or other resources are permitted but the issue should be complete and accurate and should not require the reader to visit the forum or any other platform to understand what is being described.
+    4. Issues should strive to describe the minimum acceptable condition.
+    5. Issues should focus on user tasks and avoid comparisons to other software solutions.
+    6. The user or Contributor SHOULD seek consensus on the accuracy of their observation and the value of solving the problem.
+    7. To submit a solution to a problem, a Contributor SHALL create a pull request back to the project.
+    8. Contributors and Maintainers SHALL NOT commit changes directly to the target branch.
+    9. To discuss a proposed solution, users MAY comment on the platform pull request. Forum conversations regarding the solution should be discouraged and conversation redirected to the Pull Request or the related issue.
+    10. To accept or reject a Pull Request, a Maintainer SHALL use the platform interface.
+    11. Maintainers SHOULD NOT merge their own PRs except:
+      - in exceptional cases, such as non-responsiveness from other Maintainers for an extended period (more than 1-2 days).
+      - If the maintainer is also the primary developer of the workbench or subsystem.
+
+    12. Maintainers SHALL merge valid PRs from other Contributors rapidly.
+    13. Maintainers MAY, at their discretion merge PRs that have not met all criteria to be considered valid to:
+        - (a) ending fruitless discussions
+        - (b) capturing toxic patches in the historical record
+        - (c) engaging with the Contributor on improving their patch quality.
+    14. Maintainers SHALL NOT make value judgments on correct patches.
+    15. Any Contributor who has value judgments on a PR SHOULD express these via their own PR.
+    16. The user who created an issue SHOULD close the issue after checking the PR is successful.
+    17. Maintainers SHOULD close user issues that are left open without action or update for an uncomfortable period.
```

Comment:
"uncomfortable" feels strange to me here, maybe "extended"? Or are you deliberately looking for a different word than used in the merge discussion above?

Reply:
That word also is borrowed from C4.  I initially thought to change it to 'extended' but reconsidered because that invites a definition.  What exactly does, 'extended' mean?  
How long to leave a problem around sort of depends on the nature of the problem.  I left 'Uncomfortable' because it implies that there's a judgment to be made. 
That said, I can be convinced otherwise.

Reply:
a little one..
Be consistent with capitalisation rules. User is defined, should be User.

Reply:
> I left 'Uncomfortable' because it implies that there's a judgment to be made

I might understand this wrong. As a non-native english speaker, for me 'Uncomfortable' refers to an emotion most of the time and thus is highly subjective. One person might **feel** uncomfortable after a few hours, another feels uncomfortable after a few days. I think we do need a definition whether we use extended or uncomfortable.

Reply:
> I think we do need a definition whether we use extended or uncomfortable.

Propose one.  I guarantee it won't fit all cases.  A minor issue around a GUI change that gets no attention should probably be closed in a matter of weeks.  A major issue might not be 'uncomfortable' for years.  What standard would you propose?
We should be establishing a policy that guides good judgments, not a set of rules that binds behavior.  I agree that the word 'uncomfortable' is ...odd.   It's the sentiment behind it that I agree with.  Having a hard rule on when to close issues either means the tracker gets cluttered with noise or you're prematurely closing things that shouldn't.

Reply:
> how about "unreasonable"

I like this. Upon reasoning a judgement can be make. A sentiment can not be judged, it just is.

Reply:
In the spirit of trusting the maintainer, or an alternate maintainer, why not just let he/she/them decide? The worst that can happen is the PR gets re-opened. Not a big deal.

Reply:
This is about issues, not PR's, but they can be re-opened as well, even more easily than a PR.
I know of projects here on GitHub, that have several hundreds or more open issues, some of them are ancient, meaning several years old, sometimes older than a decade. To keep track of an enhanced amount of issues, a clean, concise and continued tagging of the issues is necessary, so they can be found. If this can not be guaranteed, a way to somehow regulate closing of issues is a good thing. So I don't fear about an issue getting closed unnecessarily, but an issue not getting closed, even it's no longer valid.

Reply:
"unreasonable" is reasonable  :-)